orchestrator/graphql/utils/get_subscription_product_blocks.py

```diff
@@ -58,8 +59,11 @@ def new_product_block(item: dict[str, Any]) -> Generator:
 async def get_subscription_product_blocks(
     subscription_id: UUID, tags: Optional[list[str]] = None, product_block_instance_values: Optional[list[str]] = None
 ) -> list[ProductBlockInstance]:
-    subscription_model = SubscriptionModel.from_subscription(subscription_id)
-    subscription = build_extended_domain_model(subscription_model)
+    if cached_model := from_redis(subscription_id):
```

Comment:
might be good to do the same in `subscriptions` resolver `get_subscription_details`

Reply:
I would probably move this to a separate function, because for get_subscription_product_blocks it's not relevant where the subscription comes from. That also makes it easy to reuse it in other resolvers/functions

Reply:
I couldn't figure out how to make a query that uses `get_subscription_details()`

Also it uses the subscription as a model, not as dict. As far as I can tell this (get_subscription_product_blocks) is the only place where we need the subscription as a dict

Reply:
Made it a function